covquestions-editor-app/src/store/questionnaireInEditor.ts

```diff
@@ -63,9 +63,22 @@ export const editTestCase = createAction<{
   hasErrors: boolean;
 }>("editTestCase");
 
+export const setDuplicatedIdInformation = createAction<{
+  section: SectionType.QUESTIONS | SectionType.RESULT_CATEGORIES | SectionType.VARIABLES;
+}>("setDuplicatedIdInformation");
+
 export type QuestionnaireWrapper = {
   questionnaire: EditorQuestionnaire;
-  hasErrors: boolean;
+  errors: {
+    isJsonModeError: boolean;
+    formEditor: {
+      meta: boolean;
```

Comment:
How about moving the errors into the questionnaire property? The format inside the store is different anyway than the final json format (e.g. string representation of expressions and language and title on top level).

I haven't tested though, if this interferes with the JsonFormEditor in some way...

Advantage: we would not need to care about the error during removing or moving elements, since it would be stored locally next to the part it belongs to.

(I do not have a strong opinion, if this would be better than the current approach. For me it is also fine to merge it as is.)

Reply:
I recently tried to add properties to the questionnaire.
But then I had problems with the JSON editor. Therefore I currently use parallel structures.

But I see clearly the advantage you mentioned. We have to take a closer look at the synchronization between form and JSON editor anyway. Maybe we should take a closer second look at additional properties in the questionnaire regarding error states.